src/updater/csv_source.cpp

```diff
@@ -34,7 +34,7 @@ namespace csv
 SegmentLookupTable readSegmentValues(const std::vector<std::string> &paths)
 {
     CSVFilesParser<Segment, SpeedSource> parser(
-        1, qi::ulong_long >> ',' >> qi::ulong_long, qi::uint_ >> -(',' >> qi::double_));
+        1, qi::ulong_long >> ',' >> qi::ulong_long, qi::double_ >> ',' >> qi::double_);
```

Comment:
Making the weight column non-optional would cause some serious problems downstream. What about showing a warning if someone uses a three column CSV but still supporting the fallback method?